### ISSUE TITLE
moving text for up and down speed

### ISSUE THREAD

**Issue by pciavald** [jaagr/polybar#82]
when using the up/down speed meters, the number of characters is variable and it makes the whole bar move left and right.
is it possible to implement a fixed width of 3 characters on these meters, so that this moving number doesn't make the whole bar move left and right ?

**Comment by jaagr** [comment]
This has already been implemented and can done by setting the module option `udspeed-minwidth = 3`
I will however make the default value of this option to 3.

**Comment by jaagr** [comment]
A tmp fix for this would be to use a monospace font for the u/d speed, for example:

`label-connected = %essid% %{T3}%downspeed%%{T-}`

**Comment by pciavald** [comment]
i'm having a hard time installing fonts on my setup, especially siji, so i tried with a bunch of fonts and i'm not sure they're all monospace, anyway I guess the problem with non-monospace is that the chars don't have the same width event though there's 3 of them.
Although even with `udspeed-minwidth = 3` i don't have 3 chars.

```
[module/wifi]
type = internal/network
interface = wlp4s0
interval = 0.1
ping_interval = 10
udspeed-minwidth = 3
format-connected = <ramp-signal> <label-connected>
label-connected = %ifname% %essid% %{F#66}%local_ip% %{F#ff}%{T6}%upspeed% %downspeed%%{T-}
```

**Comment by jaagr** [comment]
Try `font-1 = fixed:size=10;0`

Not sure what you mean by the 3 chars, but the way it works is that it outputs spaces before the up/down rate. So if it's 1 KB/s it will show `xx1 KB/s` where x = whitespace.

**Comment by pciavald** [comment]
I tried with the fixed font and `udspeed-minwidth = 10`, the spacing is effectively here however it is pushed by the growing numbers.
![out 04d](https://cloud.githubusercontent.com/assets/3384097/19398636/de92a2ce-924d-11e6-9a50-1e89271bf59a.gif)

**Comment by jaagr** [comment]
Could you post your config and I'll check what's wrong

**Comment by pciavald** [comment]
```
;=====================================================
;
;   To learn more about how to configure Lemonbuddy
;   go to https://github.com/jaagr/lemonbuddy
;
;   The README contains alot of information
;
;=====================================================

[bar/bottom]
bottom = true
width = 100%
height = 20
offset-x = 0
offset-y = 0

background = #2B2E37
foreground = #ccfafafa

spacing = 1
lineheight = 1

padding-left = 0
padding-right = 2

module-margin-left = 1
module-margin-right = 2

;font-0 = unifont:size=9;1
;font-1 = tamsyn:pixelsize=10;0
;font-2 = unifont:size=6;-1

font-0 = NotoSans-Regular:size=8;0
font-1 = MaterialIcons:size=10;0
font-2 = Termsynu:size=8;-1
font-3 = FontAwesome:size=10;0
font-4 = unifont:size=6;-3
font-5 = siji:size=6;-3
font-6 = FreeMono:size=10;0

separator = |

modules-left = i3
modules-center =
modules-right = volume backlight cpu battery wifi wired

;tray-position = right

wm-restack = i3

[module/i3]
type = internal/i3
format = <label-state>

;pin-workspaces = true
;wsname-maxlen = 1
;index-sort = true

label-focused = o %index%
label-focused-background = #ee333333
label-focused-underline= #cc333333
label-focused-padding = 2

label-unfocused = %{O-1}⏺%{O-1} %index%
label-unfocused-padding = 2

label-urgent = ! %index%
label-urgent-background = #bd2c40
label-urgent-padding = 2

label-visible = x %index%
label-visible-foreground = #444
label-visible-padding = 2


[module/cpu]
type = internal/cpu
interval = 0.2

format = <label> <ramp-coreload>
label = CPU

ramp-coreload-0 = ▁
ramp-coreload-0-font = 2
ramp-coreload-0-foreground = #55aa55
ramp-coreload-1 = ▂
ramp-coreload-1-font = 2
ramp-coreload-1-foreground = #55aa55
ramp-coreload-2 = ▃
ramp-coreload-2-font = 2
ramp-coreload-2-foreground = #55aa55
ramp-coreload-3 = ▄
ramp-coreload-3-font = 2
ramp-coreload-3-foreground = #55aa55
ramp-coreload-4 = ▅
ramp-coreload-4-font = 2
ramp-coreload-4-foreground = #f5a70a
ramp-coreload-5 = ▆
ramp-coreload-5-font = 2
ramp-coreload-5-foreground = #f5a70a
ramp-coreload-6 = ▇
ramp-coreload-6-font = 2
ramp-coreload-6-foreground = #ff5555
ramp-coreload-7 = █
ramp-coreload-7-font = 2
ramp-coreload-7-foreground = #ff5555

bar-load-width = 20
bar-load-foreground-0 = #55aa55
bar-load-foreground-1 = #55aa55
bar-load-foreground-2 = #f5a70a
bar-load-foreground-3 = #ff5555
bar-load-gradient = false
bar-load-indicator = │
bar-load-indicator-font = 2
bar-load-indicator-foreground = #ff
bar-load-fill = ━
bar-load-fill-font = 2
bar-load-empty = ━
bar-load-empty-font = 2
bar-load-empty-foreground = #444444

[module/wifi]
type = internal/network
interface = wlp4s0
interval = 0.1
ping_interval = 10
udspeed-minwidth = 10

format-connected = <ramp-signal> <label-connected>
label-connected = %ifname% %essid% %{F#66}%local_ip% %{F#ff}%{T6}%upspeed% %downspeed%%{T-}
ramp-signal-0 = 
ramp-signal-1 = 
ramp-signal-2 = 
ramp-signal-3 = 
ramp-signal-4 = 

;ramp-signal-0 = 1
;ramp-signal-1 = 2
;ramp-signal-2 = 3
;ramp-signal-3 = 4
;ramp-signal-4 = 5
;ramp-signal-5 = 6

label-disconnected =  %{F#66}%ifname%
label-disconnected-foreground = #66

format-packetloss = <animation-packetloss> <label-connected>
animation-packetloss-0 = ⚠
animation-packetloss-0-foreground = #ffa64c
animation-packetloss-1 = 📶
animation-packetloss-1-foreground = #000000
animation-packetloss-framerate = 500


[module/wired]
type = internal/network
interface = enp3s0
interval = 0.1

label-connected =  %ifname% %{F#66}%local_ip% [%linkspeed%] %{F#ff}%upspeed% %downspeed%
label-disconnected = %{F#66}%ifname% 
label-disconnected-foreground = #66

[module/volume]
type = internal/volume
speaker_mixer = Speaker
headphone_mixer = Headphone
headphone_control_numid = 9

format-volume = <ramp-volume> <label-volume>
label-muted =   muted
label-muted-foreground = #66

ramp-volume-0 = 
ramp-volume-1 = 
ramp-volume-2 = 
ramp-volume-3 = 


[module/backlight]
type = internal/backlight
card = intel_backlight
format = <ramp> <label>

ramp-0 = 
ramp-1 = 
ramp-2 = 
ramp-foreground = #66


[module/battery]
type = internal/battery
full_at = 98
battery = BAT1
adapter = ADP1
poll-interval = 1

format-charging = C  <animation-charging> <label-charging>
format-discharging = D  <ramp-capacity> <label-discharging>
format-full = F  <ramp-capacity> <label-full>

ramp-capacity-0 = 
ramp-capacity-0-foreground = #f53c3c
ramp-capacity-1 = 
ramp-capacity-1-foreground = #ffa900
ramp-capacity-2 = 
ramp-capacity-3 = 
ramp-capacity-4 = 

animation-charging-0 = 
animation-charging-1 = 
animation-charging-2 = 
animation-charging-3 = 
animation-charging-4 = 
animation-charging-framerate_ms = 750


[module/memory]
type = internal/memory
format = <label> <bar-used>
label = RAM

bar-used-width = 30
bar-used-foreground-0 = #aaff77
bar-used-foreground-1 = #aaff77
bar-used-foreground-2 = #fba922
bar-used-foreground-3 = #ff5555
bar-used-indicator = |
bar-used-indicator-font = 6
bar-used-indicator-foreground = #ff
bar-used-fill = ─
bar-used-fill-font = 6
bar-used-empty = ─
bar-used-empty-font = 6
bar-used-empty-foreground = #444444


[module/powermenu]
type = custom/menu

label-open = 
label-close = 
label-separator = /

menu-0-0 = Terminate WM
menu-0-0-foreground = #e60053
menu-0-0-exec = i3-msg -t command exit
menu-0-1 = Reboot
menu-0-1-foreground = #e60053
menu-0-1-exec = menu-open-1
menu-0-2 = Power off
menu-0-2-foreground = #e60053
menu-0-2-exec = menu-open-2

menu-1-0 = Cancel
menu-1-0-foreground = #e60053
menu-1-0-exec = menu-open-0
menu-1-1 = Reboot
menu-1-1-foreground = #e60053
menu-1-1-exec = sudo reboot

menu-2-0 = Power off
menu-2-0-foreground = #e60053
menu-2-0-exec = sudo poweroff
menu-2-1 = Cancel
menu-2-1-foreground = #e60053
menu-2-1-exec = menu-open-0


[module/counter]
type = internal/counter
format = <counter>
interval = 0.1

; vim:ft=dosini

```

**Comment by jaagr** [comment]
Looks like you reference the wrong font for the u/d speed. T6 is actually siji in your config. You'd have to use T7.

**Comment by pciavald** [comment]
that's not in the man :) 
this time i'm sure it's the monospace : 
![out](https://cloud.githubusercontent.com/assets/3384097/19407279/9b4ca58c-929d-11e6-8c8b-9e72102cf164.gif)

**Comment by jaagr** [comment]
True :) It uses the same behaviour as lemonbar does. Should be documented on a page in the wiki...

Could you replace the last font with `font-6 = fixed:size=10` just to ensure that it's not a font issue.

**Comment by pciavald** [comment]
notice how the numbers are cut too ? this was on the previous capture too on the S of KB/s
![out](https://cloud.githubusercontent.com/assets/3384097/19407522/5e257cf8-92a5-11e6-9a73-17e38d8900dc.gif)

**Comment by jaagr** [comment]
hm, odd..

**Comment by Maddogruss** [comment]
I have the exact same problem, pretty much all of my modules that include changing numbers (internet speed, CPU temp, time) make the entire bar move left and right, regardless of the font.
I tried doing what was suggested here but no luck...

Here's my config:


```
[colors]
background = #222
background-alt = #444
foreground = #dfdfdf
foreground-alt = #555
primary = #ffb52a
secondary = #e60053
alert = #bd2c40

[bar/mybar]
monitor = ${env:MONITOR:}
width = 100%
height = 20
radius = 0.0
fixed-center = false
bottom = true
background = ${colors.background}
foreground = ${colors.foreground}
line-size = 2
line-color = #f00
border-size = 0
border-color = #00000000
padding-left = 2
padding-right = 2
module-margin-left = 2
module-margin-right = 2
font-0 = Cantarell:pixelsize=10;1
font-1 = fixed:pixelsize=10;1
font-2 = unifont:fontformat=truetype:size=8:antialias=false;0
font-3 = Wuncon Siji:pixelsize=10;1
modules-left = i3 mpd
modules-right = root home internal memory cpu temperature internet vpn date
tray-position = right
tray-padding = 3
scroll-up = i3wm-wsnext
scroll-down = i3wm-wsprev
cursor-click = pointer
cursor-scroll = ns-resize
separator = |

[module/i3]
type = internal/i3
pin-workspaces = true
strip-wsnumbers = false
index-sort = false
enable-click = false
enable-scroll = true
wrapping-scroll = false
reverse-scroll = true
format = <label-state> <label-mode>
label-mode = %mode%
label-mode-padding = 2
label-mode-background = #e60053
label-focused = %name%
label-focused-foreground = #ffffff
label-focused-background = #3f3f3f
label-focused-underline = #fba922
label-focused-padding = 2
label-unfocused = %name%
label-unfocused-padding = 2
label-visible = %index%
label-visible-underline = #555555
label-visible-padding = 2
label-urgent = %name%
label-urgent-foreground = #000000
label-urgent-background = #bd2c40
label-urgent-padding = 2
label-separator-padding = 2
label-separator-foreground = #ffb52a

[module/root]
type = internal/fs
mount-0 = /
interval = 1
format-mounted = <label-mounted>
label-mounted = %{F#28bed8}Root:%{F#FFFFFF}  %free%

[module/home]
type = internal/fs
mount-0 = /home
interval = 1
format-mounted = <label-mounted>
label-mounted = %{F#28bed8}Home:%{F#FFFFFF}  %free%

[module/internal]
type = internal/fs
mount-0 = /home/henry/internal
interval = 1
fixed-values = true
format-mounted = <label-mounted>
label-mounted = %{F#28bed8}Internal:%{F#FFFFFF}  %free%

[module/mpd]
type = internal/mpd
format-online = <label-song> - <label-time>
label-song = %artist% - %title%
label-song-maxlen = 64
label-song-ellipsis = true

[module/cpu]
type = internal/cpu
interval = 1
format = "<label>%"
format-prefix = " CPU: "
format-prefix-foreground = #4bffdc
label = %percentage%

[module/temperature]
type = internal/temperature
thermal-zone = 0
warn-temperature = 60
hwmon-path = /home/henry/.config/polybar/hwmon/temp1_input
format = <label>
format-prefix = " TMP: "
format-prefix-foreground = #4bffdc
format-warn = <label-warn>
format-warn-prefix = " TMP: "
format-warn-prefix-foreground = #ff0000
label = %temperature-c%
label-warn = %temperature-c%
label-warn-foreground = #ff0000

[module/memory]
type = internal/memory
interval = 1
format-prefix = " RAM: "
format-prefix-foreground = #4bffdc
label = %gb_used%

[module/internet]
type = internal/network
interface = enp6s0
interval = 1
label-connected = " E: %local_ip% ( D: %downspeed% | U: %upspeed% )"
label-connected-foreground = #55aa55
udspeed-minwidth = 3
label-disconnected = "  E: Offline"
label-disconnected-foreground = #ff0000

[module/vpn]
type = internal/network
interface = tun0
interval = 1
label-connected-foreground = #55aa55
label-connected = " VPN: On"
label-disconnected = " VPN: Off "
label-disconnected-foreground = #ff0000

[module/date]
type = internal/date
interval = 1
label = %time% %date% 
date = "%A  %d/%m/%y"
time = " %H:%M -"
format-prefix = 
format-prefix-foreground = #0a6cf5

[settings]
screenchange-reload = true
;compositing-background = xor
;compositing-background = screen
;compositing-foreground = source
;compositing-border = over

[global/wm]
margin-top = 5
margin-bottom = 5
; vim:ft=dosini
```
